src/samples/network/echo-client.c

```diff
@@ -73,9 +73,9 @@ static bool
 on_can_write(void *data, struct sol_socket *s)
 {
     int r = -1;
-    const char *value = data;
+    struct sol_buffer buffer = SOL_BUFFER_INIT_CONST(data, strlen(data));
 
-    r = sol_socket_sendmsg(sock, value, strlen(value), &address);
+    r = sol_socket_sendmsg(sock, &buffer, &address);
     if (r < 0) {
```

Comment:
for the sake of consistency, `sol_buffer_fini()` the buffer. People may copy & paste this code and will lead to them copying leaks.